os_stub/cryptlib_openssl/pk/x509.c

```diff
@@ -2300,12 +2311,25 @@ bool libspdm_set_attribute_for_req(X509_REQ *req, uint8_t *req_info, size_t req_
         return false;
     }
 
-    /*req_info sequence, all ret is ok because the req_info has been verified before*/
+    /*get subject name from req_info and set it to CSR*/
+    x509_req_info = d2i_X509_REQ_INFO(NULL, (const unsigned char **)(&req_info), req_info_len);
+    if (x509_req_info) {
+        X509_REQ_set_subject_name(req, x509_req_info->subject);
+        X509_REQ_INFO_free(x509_req_info);
+    } else {
+        return false;
+    }
+
+    /*req_info sequence, all req_info format is ok because the req_info has been verified before*/
     ret = libspdm_asn1_get_tag(&ptr, end, &obj_len,
                                LIBSPDM_CRYPTO_ASN1_SEQUENCE | LIBSPDM_CRYPTO_ASN1_CONSTRUCTED);
 
     /*integer:version*/
     ret = libspdm_asn1_get_tag(&ptr, end, &obj_len, LIBSPDM_CRYPTO_ASN1_INTEGER);
+    /*check req_info verson. spec PKCS#10: It shall be 0 for this version of the standard.*/
+    if (*ptr != 0) {
```

Comment:
Do you assume obj_len is 1 ?
If so, you need add check such.

Reply:
Thanks. I have added the obj_len check:  `if ((obj_len != 1) || (*ptr != 0)) {`